src/reconcile/rec_write.c

```diff
@@ -1580,18 +1584,6 @@ __rec_child_deleted(
 	}
 
 	/*
-	 * If there are deleted child pages that we can't discard immediately,
-	 * keep the page dirty so they are eventually freed.
-	 */
-	if (ref->addr != NULL) {
-		r->leave_dirty = 1;
-
-		/* This page cannot be evicted, quit now. */
-		if (F_ISSET(r, WT_EVICTING))
-			return (EBUSY);
-	}
-
-	/*
```

Comment:
Continuing to evict seems wrong to me: we're going to throw away version information, so future readers can't figure out whether to see the deleted data or not.